nose2/plugins/junitxml.py

```diff
@@ -143,6 +143,10 @@ def testOutcome(self, event):
             test_args = ':'.join(testid_lines[1:])
             method = '%s (%s)' % (method, test_args)
 
+        # subtests do not report success
+        if event.outcome == result.SUBTEST and event.exc_info is None:
```

Comment:
Yeah the check happens twice in junitxml. This is due to the plugin design and to avoid creating empty `<subtest>` tags. It looks far more elegant in the result plugin (see below). Also note that we need exc_info around to tell errors from failures. I do not consider this a DRY violation that would require action. I like "explicit" and I like "grep-able". I also don't mind typing. 😃 

Subtests do not create additional tests, they only create additional _test results_ (and only if something goes wrong). One could argue that there is no such thing as a passing subtest (it is the enclosing test that passes), and I'd rather not introduce the concept.

Reply:
Okay, I'm persuaded by this not to try to zhuzh up the Event interface. The fact that we're following the example set by unittest supports your call on this too.